Mathlib/Algebra/Category/Ring/Constructions.lean

```diff
@@ -215,6 +216,50 @@ set_option linter.uppercaseLean3 false in
 
 end Product
 
+section Pi
+
+variable {ι : Type u} (R : ι → CommRingCat.{u})
+
+/--
+The categorical product of rings is the cartesian product of rings. This is its `Fan`.
+-/
+@[simps! pt]
+def piFan : Fan R :=
+  Fan.mk (CommRingCat.of ((i : ι) → R i)) fun i ↦
+    { toFun := fun g ↦ g i
+      map_one' := by aesop
+      map_mul' := by aesop
+      map_zero' := by aesop
+      map_add' := by aesop }
```

Comment:
I think this already exists as `RingHom.eval` or similar?

Reply:
This is `Pi.evalRingHom`.